pkg/kp/rcstore/consul_store.go

```diff
@@ -138,31 +138,24 @@ func (s *consulStore) List() ([]fields.RC, error) {
 func (s *consulStore) WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan error) {
 	outCh := make(chan []fields.RC)
 	errCh := make(chan error)
+	inCh := make(chan api.KVPairs)
+
+	go consulutil.WatchPrefix(kp.RC_TREE, s.kv, inCh, quit, errCh)
```

Comment:
It's a little more complex than this unfortunately.  This watcher might still be writing errors to the error channel when the channel is closed below.